src/PgRvn.Server/RqlQuery.cs

```diff
@@ -0,0 +1,428 @@
+
+            if (_result.Includes.Count > 0 ||
+                sample.TryGet("@metadata", out BlittableJsonReaderObject metadata) && metadata.TryGet("@id", out string _))
+            {
+                _hasId = true;
+                Columns["id()"] = new PgColumn
+                {
+                    Name = "id()",
+                    FormatCode = resultsFormat,
+                    TypeModifier = -1,
+                    TypeObjectId = PgTypeOIDs.Text,
+                    DataTypeSize = -1,
+                    ColumnIndex = 0,
+                    TableObjectId = 0
+                };
+            }
+
+            BlittableJsonReaderObject.PropertyDetails prop = default;
+            for (int i = 0; i < sample.Count; i++)
+            {
+                sample.GetPropertyByIndex(i, ref prop);
+                if (prop.Name == "@metadata")
+                    continue;
+
+                var (type, size) = (prop.Token & BlittableJsonReaderBase.TypesMask) switch
+                {
+                    BlittableJsonToken.CompressedString => (PgTypeOIDs.Text, -1),
+                    BlittableJsonToken.String => (PgTypeOIDs.Text, -1),
+                    BlittableJsonToken.Boolean => (PgTypeOIDs.Bool, PgConfig.TrueBuffer.Length),
+                    BlittableJsonToken.EmbeddedBlittable => (PgTypeOIDs.Json, -1),
+                    BlittableJsonToken.Integer => (PgTypeOIDs.Int8, sizeof(long)),
+                    BlittableJsonToken.LazyNumber => (PgTypeOIDs.Float8, sizeof(double)),
+                    BlittableJsonToken.Null => (PgTypeOIDs.Json, -1),
+                    BlittableJsonToken.StartArray => (PgTypeOIDs.Json, -1),
+                    BlittableJsonToken.StartObject => (PgTypeOIDs.Json, -1),
+                    _ => throw new NotSupportedException()
+                };
+
+                string val = (prop.Token & BlittableJsonReaderBase.TypesMask) switch
+                {
+                    BlittableJsonToken.CompressedString => (string)prop.Value,
+                    BlittableJsonToken.String => (LazyStringValue)prop.Value,
+                    _ => null
+                };
+
+                if (val != null && TryConvertStringValue(val, out object output))
+                {
+                    switch (output)
+                    {
+                        case DateTime dt:
+                            size = 8;
+                            type = PgTypeOIDs.Timestamp;
+                            if (dt.Kind == DateTimeKind.Utc)
+                                type = PgTypeOIDs.TimestampTz;
+                            break;
+
+                        case DateTimeOffset:
+                            type = PgTypeOIDs.TimestampTz;
+                            size = 8;
+                            break;
+
+                        case TimeSpan:
+                            type = PgTypeOIDs.Interval;
+                            size = 16;
```

Comment:
Let's move the hard coded sizes to constants with good names

Reply:
Do you think we should create a class for every type we use and do: `PgInt8.Oid`, `PgInt8.Size` and for conversions handle them in there with static functions: `PgInt8.ToBytes(..)`, `int x =PgInt8.FromBytes(..)`?

Reply:
that would be nice I think

Reply:
Done